erts/emulator/beam/jit/x86/beam_asm.cpp

```diff
@@ -800,8 +800,6 @@ extern "C"
 
     void beamasm_purge_module(const void *native_module_exec,
                               void *native_module_rw) {
-        ASSERT(native_module_exec != native_module_rw);
```

Comment:
For  this one, I can also wrap it in ifdef/endif for `HAVE_LINUX_PERF_SUPPORT` and dump/map support enabled, although it seem to complicate the assert unnecessarily.

Reply:
I agree that we can remove it, maybe it will be added later if we get into problems with it again.